plugins/spectate.cpp

```diff
@@ -523,6 +530,9 @@ static void follow_a_dwarf(color_ostream &out) {
     intervals.push_back(0);
     add_bucket(citizen_combat_units, units, intervals, weights, config.prefer_conflict ? CITIZEN_COMBAT_PREFERRED_WEIGHT : JOB_WEIGHT);
     add_bucket(other_combat_units, units, intervals, weights, config.prefer_conflict ? OTHER_COMBAT_PREFERRED_WEIGHT : JOB_WEIGHT);
+    if (config.prefer_nicknamed) {
+        add_bucket(nicknamed_units, units, intervals, weights, NICKNAMED_CITIZEN_PREFERRED_WEIGHT);
+    }
```

Comment:
This effectively removes nicknamed dwarves from the selection pool if `prefer_nicknamed` is false.

These code reviews take up a lot of my time. Please be sure to test your code carefully.